examples/src/Auction/Offchain.hs

```diff
@@ -84,6 +86,7 @@ txBid submitter offerOref bid = do
       Just deadline = A.getBidDeadline datum
       seller = A.getSeller datum
       lotPlusPreviousBidPlusNft = outputValue output
+  validityInterval <- slotRangeBefore deadline
```

Comment:
This validity interval is stricter than necessary. Shouldn't the deadline be included?

Reply:
> and we have no control on how the ledger translates the slot range of the transaction into this millisecond range

Does it not depend on the `SlotConfig` alone? Otherwise, your explanation makes sense.

Reply:
Following your reference, [this is where the time interval is computed](https://github.com/input-output-hk/cardano-ledger/blob/38aa337834ccb9b5f47dbf9e81a3b986dcdd57f5/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs#L309-L315).

It makes the least surprising choices, that is, what I would guess it should do.

If the meaning of slot changes from era to era, off-chain code and tools will have to be adapted. There would be little incentive to do it frequently.

Reply:
If we want to be as precise as we can `slotRangeBefore` should return the range of slots ending before or at the given time.

In this way, if the deadline is at the end of a slot, it gets included.

I don't think this needs to block this PR though.